ecommerce/extensions/payment/utils.py

```
@@ -53,3 +63,76 @@ def clean_field_value(value):
         A cleaned string.
     """
     return re.sub(r'[\^:"\']', '', value)
+
+
+class SDNClient(object):
+    """A utility class that handles SDN related operations."""
+    def __init__(self, api_url, api_key, sdn_list):
+        self.api_url = api_url
+        self.api_key = api_key
+        self.sdn_list = sdn_list
+
+    def search(self, name, country):
+        """
+        Searches the OFAC list for an individual with the specified details.
+        The check returns zero hits if:
+            * request to the SDN API times out
+            * SDN API returns a non-200 status code response
+            * user is not found on the SDN list
+
+        Args:
+            name (str): Individual's full name.
+            country (str): ISO 3166-1 alpha-2 country code where the individual is from.
+        Returns:
+            json: SDN API response.
+        """
+
+        params = urlencode({
+            'sources': self.sdn_list,
+            'api_key': self.api_key,
+            'type': 'individual',
+            'name': name,
+            'countries': country
+        })
+        sdn_check_url = '{api_url}?{params}'.format(
+            api_url=self.api_url,
+            params=params
+        )
+
+        try:
+            response = requests.get(sdn_check_url, timeout=settings.SDN_CHECK_REQUEST_TIMEOUT)
+        except requests.exceptions.Timeout:
+            logger.warning('Connection to US Treasury SDN API timed out for [%s].', name)
+            return None
+
+        if response.status_code != 200:
```

Ideally, you should raise an exception here and above. We should avoid special return values where possible. In the past I've established a base exception class (e.g. `SDNClientError`) and caught that. Given that the response to a timeout and a server error will be the same, you can differentiate with an error message rather than subclasses. If, however, we chose to retry a timeout, you'd want to make an appropriate exception subclass.